src/ops/kernels/embedding_kernels.cpp

```diff
@@ -388,177 +510,53 @@ void Embedding::backward_kernel(TI const *input_ptr,
-                                 output.domain.get_volume(),
-                                 stream);
-    } else if (m->output_type[0] == DT_FLOAT) {
-      Embedding::backward_kernel(input.get_int64_ptr(),
-                                 output.get_float_ptr(),
-                                 weight_grad.get_float_ptr(),
-                                 in_dim,
-                                 out_dim,
-                                 batch_size,
-                                 m->aggr,
-                                 output.domain.get_volume(),
-                                 stream);
-    } else if (m->output_type[0] == DT_DOUBLE) {
-      Embedding::backward_kernel(input.get_int64_ptr(),
-                                 output.get_double_ptr(),
-                                 weight_grad.get_double_ptr(),
-                                 in_dim,
-                                 out_dim,
-                                 batch_size,
-                                 m->aggr,
-                                 output.domain.get_volume(),
-                                 stream);
-    } else {
-      assert(false && "Unsupported DataType in Embedding");
-    }
-  }
-
-  if (m->profiling) {
-    checkCUDA(hipDeviceSynchronize());
-    // print_tensor<float>(output_grad_ptr, output_grad_domain.volume(),
-    // "[Embedding:backward:output_grad]"); print_tensor<float>(kernel_grad_ptr,
-    // kernel_grad_domain.get_volume(), "[Embedding:backward:weight_grad]");
-    // print_tensor<TI>(input_ptr, input_domain.get_volume(),
-    // "[Embedding:backward:input]");
-  }
-}
-
 template <typename TD>
 __global__ void rand_generate_int(TD *ptr, size_t size, TD p) {
   CUDA_KERNEL_LOOP(i, size) {
     ptr[i] = i % p;
   }
 }
 
-void Embedding::rand_generate_int64_wrapper(int64_t *ptr,
-                                            size_t size,
-                                            int64_t p) const {
-  hipStream_t stream;
-  checkCUDA(get_legion_stream(&stream));
-  // Randomly initialize the intput tensor to avoid out of index range issues
-  hipLaunchKernelGGL(rand_generate_int,
-                     GET_BLOCKS(size),
-                     CUDA_NUM_THREADS,
-                     0,
-                     stream,
-                     ptr,
-                     size,
-                     p);
-}
-
-void Embedding::rand_generate_int32_wrapper(int32_t *ptr,
-                                            size_t size,
-                                            int32_t p) const {
-  hipStream_t stream;
-  checkCUDA(get_legion_stream(&stream));
-  // Randomly initialize the intput tensor to avoid out of index range issues
-  hipLaunchKernelGGL(rand_generate_int,
-                     GET_BLOCKS(size),
-                     CUDA_NUM_THREADS,
-                     0,
-                     stream,
-                     ptr,
-                     size,
-                     p);
-}
-
 #ifdef DEADCODE
```

Comment:
Can this deadcode be safely removed?

Reply:
Not sure how this deadcode was being used. In the current flow its not being used, kept it just to maintain backward compatibility.

Reply:
#570